smalldata_tools/ana_funcs/dropletCode/greedyguess.py

```diff
@@ -1,40 +1,56 @@
 import numpy as np
 #Greedy assignment of photons
 #from fitdrop import placephots
-#from numba import jit
+from numba import jit
 
-#@jit
+@jit(nopython=True, cache=True)
 def greedyguess(img,nophots,aduspphot):
-    #relative indices for the 4 tiles, order is carefully chosen
-    xs=np.array([[0,0],[0,0],[-1,0],[1,0]])
-    ys=np.array([[-1,0],[1,0],[0,0],[0,0]])
-
-    timg=img.copy()
     pxs=np.zeros(nophots)
     pys=pxs.copy()
-    for n in range(nophots):
+    
+    fimg=img.copy()/aduspphot
+    pimg=fimg//1
+    timg=fimg-pimg
+
+    nophots_assigned=0
+    #pimg->sparse array, fill pxs&pys, figure out what to do with photons > 1.
+    for ii in range(pimg.shape[0]):
+        for jj in range(pimg.shape[1]):
+            for photon in range(pimg[ii,jj]):
+                pxs[nophots_assigned] = ii
+                pys[nophots_assigned] = jj
+                nophots_assigned=nophots_assigned+1
+
+    for n in range(nophots_assigned,nophots):
         #find max intensity for next photon guess
-        i,j=np.unravel_index(np.argmax(timg),timg.shape)
+        maximg=-1
+        for ii in range(timg.shape[0]):
```

Comment:
I assume this weird way of finding the max is because of jit and the impossibility to use numpy function here?

Reply:
Yes, numba did not really like the unravel_index and as I'm compiling this with jit, I should not have the python-loop penalty. It is possible that I could have tried harder to make number like unravel_index, but I assume that it does not matter much how I'm making this loop efficient.